gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatch.java

```diff
@@ -0,0 +1,230 @@
+   * DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Zone> createZone(ZoneInfo zone, Dns.ZoneOption... options) {
+    DnsBatchResult<Zone> result = new DnsBatchResult<>();
+    DnsRpc.Callback<ManagedZone> callback = createZoneCallback(this.options, result);
+    Map<DnsRpc.Option, ?> optionMap = optionMap(options);
+    batch.addCreateZone(zone.toPb(), callback, optionMap);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "delete zone" operation to this batch. The returned {@link
+   * DnsBatchResult} will return {@code true} upon calling {@link DnsBatchResult#get()} on
+   * successful deletion, {@code false} if the zone was not found, or it will throw a {@link
+   * DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Boolean> deleteZone(String zoneName) {
+    DnsBatchResult<Boolean> result = new DnsBatchResult<>();
+    DnsRpc.Callback<Void> callback = createDeleteZoneCallback(result);
+    batch.addDeleteZone(zoneName, callback);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "get zone" operation to this batch. The {@code options} can be
+   * used to restrict the fields returned in the same way as for {@link Dns#getZone(String,
+   * Dns.ZoneOption...)}. The returned {@link DnsBatchResult} will return the requested {@link Zone}
+   * upon calling {@link DnsBatchResult#get()} on successful completion, {@code null} if no such
+   * zone exists, or it will throw a {@link DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Zone> getZone(String zoneName, Dns.ZoneOption... options) {
+    DnsBatchResult<Zone> result = new DnsBatchResult<>();
+    DnsRpc.Callback<ManagedZone> callback = createZoneCallback(this.options, result);
+    Map<DnsRpc.Option, ?> optionMap = optionMap(options);
+    batch.addGetZone(zoneName, callback, optionMap);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "get project" operation to this batch. The {@code options} can
+   * be used to restrict the fields returned in the same way as for {@link
+   * Dns#getProject(Dns.ProjectOption...)}. The returned {@link DnsBatchResult} will return the
+   * requested {@link ProjectInfo} upon calling {@link DnsBatchResult#get()} on successful
+   * completion, or it will throw a {@link DnsException} if the operation failed.
+   */
+  public DnsBatchResult<ProjectInfo> getProject(Dns.ProjectOption... options) {
+    DnsBatchResult<ProjectInfo> result = new DnsBatchResult<>();
+    DnsRpc.Callback<Project> callback = createProjectCallback(result);
+    Map<DnsRpc.Option, ?> optionMap = optionMap(options);
+    batch.addGetProject(callback, optionMap);
+    return result;
+  }
+
+  // todo(mderka) implement remaining operations
+
+  /**
+   * Submits this batch for processing using a single HTTP request.
+   */
+  public void submit() {
+    batch.submit();
+  }
+
+  private Map<DnsRpc.Option, ?> optionMap(Option... options) {
```

Comment:
instead of copy/paste make use the one in DnsImpl (and make it package scope and static)?